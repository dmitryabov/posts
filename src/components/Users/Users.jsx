import React from 'react';
import style from './Users.module.css';
import User from './User/User';

const Users = (props) => {
    return (
        <div>
          <h2>Responsive "Meet The Team" Section</h2>
          <p>Resize the browser window to see the effect.</p>
          <br/>
          <div className={style.row}>
          {props.users.map( user => <User key={user.id} user={user}/> )}
          </div>
        </div>
    )
}

export default Users;