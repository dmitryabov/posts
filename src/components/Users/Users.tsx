import React from 'react';
import style from './Users.module.css';
import User from './User/User';
import {UsersType} from '../../redux/users-reduser';


type PropsType = {
  users: Array<UsersType>
}


const Users: React.FC<PropsType> = (props) => {
    return (
        <div>
          <h2>List of users</h2>
          <br/>
          <div className={style.row}>
          {props.users.map( user => <User key={user.id} user={user}/> )}
          </div>
        </div>
    )
}

export default Users;