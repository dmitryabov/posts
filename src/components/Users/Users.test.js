import React from "react";
import renderer from "react-test-renderer";
import Users from "./Users";


const users = [
  {
    "id": 1,
    "name": "Leanne Graham",
    "username": "Bret",
    "email": "Sincere@april.biz",
    "address": {
      "street": "Kulas Light",
      "suite": "Apt. 556",
      "city": "Gwenborough",
      "zipcode": "92998-3874",
      "geo": {
        "lat": "-37.3159",
        "lng": "81.1496"
      }
    },
    "phone": "1-770-736-8031 x56442",
    "website": "hildegard.org",
    "company": {
      "name": "Romaguera-Crona",
      "catchPhrase": "Multi-layered client-server neural-net",
      "bs": "harness real-time e-markets"
    }
  }
];


it(`Render Users`, () => {
  const tree = renderer
  .create(<Users
    isFetching={true}
    users={users}
    
  />, {
    createNodeMock: ()=>{
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
