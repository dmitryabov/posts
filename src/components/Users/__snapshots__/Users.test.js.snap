// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Users 1`] = `
<div>
  <h2>
    List of users
  </h2>
  <br />
  <div
    className="row"
  >
    <div
      className="column"
    >
      <div
        className="card"
      >
        <div
          className="container"
        >
          <h2>
            Leanne Graham
          </h2>
          <p>
            <span
              className="title"
            >
              email:
            </span>
             
            Sincere@april.biz
          </p>
          <p>
            <span
              className="title"
            >
              address:
            </span>
          </p>
          <p>
            <span
              className="title details"
            >
              street:
            </span>
             
            Kulas Light
          </p>
          <p>
            <span
              className="title details"
            >
              suite:
            </span>
             
            Apt. 556
          </p>
          <p>
            <span
              className="title details"
            >
              city:
            </span>
             
            Gwenborough
          </p>
          <p>
            <span
              className="title details"
            >
              zipcode:
            </span>
             
            92998-3874
          </p>
          <p>
            <span
              className="title"
            >
              phone:
            </span>
             
            1-770-736-8031 x56442
          </p>
          <p>
            <span
              className="title"
            >
              website:
            </span>
             
            hildegard.org
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
